README.md

```diff
@@ -73,6 +75,103 @@ Run your `crystal spec`
 crystal spec
 ```
 
+## Controller Tests
+
+
+```crystal
+class PostControllerTest < GarnetSpec::Controller::Test
+  getter handler : Amber::Pipe::Pipeline
+
+  def initialize
+    @handler = Amber::Pipe::Pipeline.new
+    @handler.build :web do
+      plug Amber::Pipe::Error.new
+      plug Amber::Pipe::Session.new
+      plug Amber::Pipe::Flash.new
+    end
+    @handler.prepare_pipelines
+  end
+end
```

Comment:
I have a question: in reply to an issue here https://github.com/amberframework/garnet-spec/issues/10#issuecomment-379771376 you suggested a little bit different approach to building controller test class:

```crystal
class BlogControllerTest < GarnetSpec::Controller::Test
  getter handler : Amber::Pipe::Pipeline
  def initialize
    @handler = Amber::Server.instance.handler
    @handler.prepare_pipelines
  end
end
```

Isn't it better to use real handler from Amber? I played around with controller specs just yesterday and to me it felt better do to it this way, because for example I changed the pipelane in `routes.cr` and forgot to change it in tests. Also, if I had multitude of tests it would be painful to have to change every single one of them after changing routes.

Reply:
I agree to an extent but there is a couple of reasons why we decided to do it this way. 

1. To decouple Garnet Spec from amber framework, if we do it the way you suggest above Garnet Spec can only be used with amber
2. CSRF Handler causes specs to fail and the GarnetSpec::Controller::Test allows you to define the pipeline without certain handlers and the dev has complete control over the pipeline. 

I just had the thought of do the following and it would prevent redefining the pipeline all completely but something does not look right

```crystal
pipeline :web do
    plug Amber::Pipe::PoweredByAmber.new
    # plug Amber::Pipe::ClientIp.new(["X-Forwarded-For"])
    plug Amber::Pipe::Error.new
    plug Amber::Pipe::Logger.new if Amber.env.production?
    plug Amber::Pipe::CSRF.new if Amber.env.production?
    plug Amber::Pipe::Session.new
    plug Amber::Pipe::Flash.new
    plug Amber::Pipe::Reload.new if Amber.env.development?
  end
```

Reply:
Thanks for the reply. That makes a lot of sense.

Reply:
An explicit link between garnet and amber doesn't need to be made in order to accomplish this. Garnet is a utility library, Amber should be able to generate specs which utilize it and explicitly link to the production pipeline. 

If there is some reason that csrf doesn't work in specs, that should be addressed with the garnet framework, or the way that amber uses garnet. Disabling CSRF is a workaround and is going to cause a bug in some application because the test application is different from the production application.

Reply:
@robacarp the PR #16 does this by only needing to specify the `HANDLER=Amber::Server.instance.handler` constant in  Amber spec_helper.cr which is simpler than the current implementation

```crystal
module GarnetSpec
  # As you can see there is no need to define a Mock class
  HANDLER = Amber::Server.instance.handler
  # Needed since the server needs to prepare the linked list for each of the pipelines
  HANDLER.prepare_pipelines
end
```